### DISCUSSION TITLE
New node names and format for import/export in mdx 2.0.0.next.9

### DISCUSSION THREAD

**Discussion by iAdramelk** [mdx-js/mdx#1500] [Q&A] [answered]
`mdx` v1 had import and export nodes. E. g.:

```
{
  type: "export",
  default: false,
  value: `export const ${name} = ${stringifyObject(headers)}`
}
```

next.9 version had new parser that changed these nodes format to `mdxjsEsm`. But just changing node type to `mdxjsEsm` didn't seems to work. Do I understand correctly that if I want to create new export node inside remark or `rehype` plugin I need to parse export string with esprima and write results to `data` field or is is done in some other way?

And if it indeed requires esprima `ast` in data field is there an easier way to create such nodes then working with the parser directly?

**Comment by wooorm** [comment, marked as answer]
It’s not an esprima AST: it’s an estree, used by acorn/eslint/and much of the ecosystem.
`mdxjsEsm` also have a `data.estree` field of that AST.
Those ASTs are used here so that we don’t have to parse the input *again*. And so that you don‘t have to deal with escaping issues.

See https://github.com/remcohaszing/remark-mdx-images and https://github.com/remcohaszing/remark-mdx-frontmatter (the latter is closes to what you’re doing)

**Reply by wooorm** [reply]
To add, you shouldn’t need to use esprima/acorn to parse something: you can add an AST directly.

For docs on this, that’s still on the todo list

**Reply by iAdramelk** [reply]
Thanks, its pretty much that I was looking for.
